File: src/main/kotlin/com/practiceKotlin/lec03/main.java
package com.practiceKotlin.lec03;

import com.practiceKotlin.lec02.Person;

public class main {
    public static void main(String[] args) {

    }

    public static void printAgeIfPerson(Object obj){
        if (obj instanceof Person) {
            Person person = (Person) obj;
            System.out.println(person.getName());
        }
    }

}

File: src/main/kotlin/com/practiceKotlin/lec03/main.kt
package com.practiceKotlin.lec03


fun main(){
    printAgeIfPerson(Person("", 100))

    val person = Person("최태현", 100)

    //java
    println(String.format("이름 : %s", person.name))
    //kotlin
    println("이름 : ${person.name}")

    val name = "최태현"
    println("이름 : $name")

    val str = """
        ABC
        EFG
        ${name}
    """.trimIndent()
    println(str)

    val str2 = "ABC"
    println(str2[0])
    println(str2[2])
}

fun printAgeIfPerson(obj: Any?){
    val person = obj as? Person     //obj가 null이 아니라면 Person
    println(person?.age);
}
